luminex/src/org/labkey/luminex/LuminexExcelParser.java

```diff
@@ -242,6 +253,93 @@ private void parseFile() throws ExperimentException
+                HashMap<String, Integer> mini = new HashMap<>();
+
+                // Check for single-file titrations. Does not check for dilution counts.
+                if (potentialTitrationCounts.get(desc) != null)
+                {
+                    // To Reviewer: It's possible to make the runtime better by removing the description from consideration in the main function's potentialTitration(Raw or Summary)Counts
+                    // once we recognize that it is a titration. However I found this makes debugging a bit harder / perhaps the code less readable because the description count is
+                    // no longer there/updated and accurate to the data.
+                    mini.put(analyteName, potentialTitrationCounts.get(desc));
+                    crossFilePT.put(desc, mini);
+                    if ((type.contentEquals("raw")) && ((potentialTitrationCounts.get(desc)) >= LuminexDataHandler.MINIMUM_TITRATION_RAW_COUNT))
+                    {
+                        _titrations.put(desc, potentialTitrations.get(desc));
+                    }
+                    else if ((type.contentEquals("summary")) && ((potentialTitrationCounts.get(desc)) >= LuminexDataHandler.MINIMUM_TITRATION_SUMMARY_COUNT))
+                    {
+                        _titrations.put(desc, potentialTitrations.get(desc));
+                    }
+
+                    if (potentialTitrations.get(desc).isQcControl() )
+                    {
+                        if ((type.contentEquals("raw") && (potentialTitrationCounts.get(desc) <= LuminexDataHandler.SINGLE_POINT_CONTROL_RAW_COUNT)) ||
+                            (type.contentEquals("summary") && (potentialTitrationCounts.get(desc) == LuminexDataHandler.SINGLE_POINT_CONTROL_SUMMARY_COUNT)))
+                        {
+                            _singlePointControls.put(desc, new SinglePointControl(potentialTitrations.get(desc)));
+                        }
+                    }
+                }
+            }
+        }
+        return crossFilePT;
+    }
+
+    private void checkCrossFilePTs(Map<String, HashMap<String, Integer>> crossFilePT, Map<String, Titration> crossFilePTs, Map<String, HashSet<Double>> dilutionCounts, String type)
+    {
+        int minimum_titration_count = (type.contentEquals("raw")) ? LuminexDataHandler.MINIMUM_TITRATION_RAW_COUNT : LuminexDataHandler.MINIMUM_TITRATION_SUMMARY_COUNT;
+        int single_point_control_count = (type.contentEquals("raw")) ? LuminexDataHandler.SINGLE_POINT_CONTROL_RAW_COUNT : LuminexDataHandler.SINGLE_POINT_CONTROL_SUMMARY_COUNT;
+
+        for (String desc : crossFilePTs.keySet())
+        {
+            if (crossFilePT.containsKey(desc))
+            {
+                for (String analyte : crossFilePT.get(desc).keySet())
+                {
+                    boolean foundTitration = false;
+                    if ((crossFilePT.get(desc).get(analyte) >= minimum_titration_count) && (dilutionCounts.get(desc).size() >= 3))

```

Comment:
I believe in the cross plate titration check we only want to be considering the distinct dilution count for the given description values. So I think this if statement can remove the first clause and just check that `dilutionCounts.get(desc).size() >= 3`. Also, I think instead of hardcoding that value to 3, we should move this to a constant like the `MINIMUM_TITRATION_RAW_COUNT`

Reply:
I've added a MINIMUM_TITRATION_DILUTION_COUNT.
I've also taken out the first clause of the if statement -- and just to verify, that change implies that for cross-file potential titrations, we're only counting the dilution counts associated with the descriptions, and not the description counts across files.
If that is the case, then we don't need to be fully populating the crossFilePTs, just dilutionCounts.

Reply:
My take is that yes, cross plate titrations are found when a description exists in multiple files at 3 or more distinct dilutions, but single plate titrations are found when the description value exists more then 5 times in a single file. We should get buy in on this from Hannah (and from DHVI).